task-driver/integration/tests/settle_match.rs

```diff
@@ -0,0 +1,391 @@
+            amount: 200,
+        },
+    }
+}
+
+/// Setup a wallet with a given order and balance
+///
+/// Returns the wallet, a blinder seed and a share seed
+async fn setup_wallet_with_order_balance(
+    order: Order,
+    balance: Balance,
+    test_args: IntegrationTestArgs,
+) -> Result<(Wallet, Scalar, Scalar)> {
+    let client = &test_args.starknet_client;
+    let (mut wallet, blinder_seed, share_seed) = new_wallet_in_darkpool(client).await?;
+
+    // Deposit the balance into the wallet
+    let old_wallet = wallet.clone();
+    wallet
+        .balances
+        .insert(balance.mint.clone(), balance.clone());
+    wallet.reblind_wallet();
+
+    // Increase the deposit allowance to transfer the balance
+    increase_erc20_allowance(
+        balance.amount,
+        &balance.mint.to_str_radix(16 /* radix */),
+        test_args.clone(),
+    )
+    .await?;
+
+    execute_wallet_update(
+        old_wallet,
+        wallet.clone(),
+        Some(ExternalTransfer {
+            mint: balance.mint,
+            amount: balance.amount.into(),
+            account_addr: biguint_from_hex_string(&test_args.account_addr),
+            direction: ExternalTransferDirection::Deposit,
+        }),
+        test_args.clone(),
+    )
+    .await?;
+
+    // Add the order to the wallet
+    let old_wallet = wallet.clone();
+    wallet.orders.insert(Uuid::new_v4(), order);
+    wallet.reblind_wallet();
+
+    execute_wallet_update(
+        old_wallet,
+        wallet.clone(),
+        None, /* transfer */
+        test_args.clone(),
+    )
+    .await?;
+
+    Ok((wallet, blinder_seed, share_seed))
+}
+
+/// Setup a buy side wallet
+async fn setup_buy_side_wallet(test_args: IntegrationTestArgs) -> Result<(Wallet, Scalar, Scalar)> {
+    setup_wallet_with_order_balance(
+        dummy_order(OrderSide::Buy, &test_args),
+        dummy_balance(OrderSide::Buy, &test_args),
+        test_args,
+    )
+    .await
+}
+
+/// Setup a sell side wallet
+async fn setup_sell_side_wallet(
+    test_args: IntegrationTestArgs,
+) -> Result<(Wallet, Scalar, Scalar)> {
+    setup_wallet_with_order_balance(
+        dummy_order(OrderSide::Sell, &test_args),
+        dummy_balance(OrderSide::Sell, &test_args),
+        test_args,
+    )
+    .await
+}
+
+/// Setup a `HandshakeResult` by mocking out the process taken by the handshake
+/// manager
+async fn setup_match_result(
+    wallet1: &Wallet,
+    wallet2: &Wallet,
+    test_args: &IntegrationTestArgs,
+) -> Result<HandshakeResult> {
+    let mut rng = thread_rng();
+    let price = FixedPoint::from_f64_round_down(EXECUTION_PRICE);
+    let base_amount = 10;
+    let quote_amount = price * Scalar::from(base_amount);
+
+    let match_ = MatchResult {
+        quote_mint: biguint_from_hex_string(&test_args.erc20_addr),
+        base_mint: biguint_from_hex_string(WETH_ADDR),
+        quote_amount: scalar_to_u64(&quote_amount.floor()),
+        base_amount,
+        direction: 0,
```

Comment:
perhaps worth documenting that this method expects party 0 in the match to be buy-side?

Reply:
changed it to be based on the arguments passed in